### PR TITLE
let "show hosts" can show git info sha of graph/meta/storage

### PR DESCRIPTION
**Pull request by liuyu85cn**
add three parameters for show hosts 

```
(root@Nebula-Console) [()]> show hosts graph
==================================================================
|  Host       |  Port  |  Status    |  role     |  git_info_sha  |
==================================================================
|  "0.0.0.0"  |  6510  |  "ONLINE"  |  "GRAPH"  |  "c41c305"     |
------------------------------------------------------------------

(root@Nebula-Console) [()]> show hosts meta
=====================================================================
|  Host           |  Port  |  Status    |  role    |  git_info_sha  |
=====================================================================
|  "127.0.0.1"  |  6500  |  "ONLINE"  |  "META"  |  "44f785b9"    |
---------------------------------------------------------------------

(root@Nebula-Console) [()]> show hosts storage
========================================================================
|  Host           |  Port  |  Status    |  role       |  git_info_sha  |
========================================================================
|  "127.0.0.1"  |  6520  |  "ONLINE"  |  "STORAGE"  |  "44f785b9"    |
------------------------------------------------------------------------
|  "127.0.0.1"  |  6530  |  "ONLINE"  |  "STORAGE"  |  "44f785b9"    |
------------------------------------------------------------------------
|  "127.0.0.1"  |  6540  |  "ONLINE"  |  "STORAGE"  |  "44f785b9"    |
------------------------------------------------------------------------
```

### CODE REVIEW

**Review by critical27 — APPROVED** [review]

File: src/executor/admin/ShowHostsExecutor.cpp [hunk 1]
```diff
@@ -18,64 +18,86 @@ folly::Future<Status> ShowHostsExecutor::execute() {
...
+            std::stringstream leaders;
+            std::stringstream parts;
+            std::size_t i = 0;
+            for (const auto &l : host.get_leader_parts()) {
+                leaders << l.first << ":" << l.second.size();
+                if (i < host.get_leader_parts().size() - 1) {
+                    leaders << kPartitionDelimeter;
+                }
+                ++i;
+            }
```
Comment (yixinglu) [inline src/executor/admin/ShowHostsExecutor.cpp:49]: could use `folly::join` to join string list

Comment (liuyu85cn) [reply src/executor/admin/ShowHostsExecutor.cpp:49]: I think folly::join is good for a well prepared, static vector.
But this logical is a little bit complex for join to handle. 
WDYT?

File: src/executor/admin/ShowHostsExecutor.cpp [hunk 2]
```diff
@@ -18,64 +18,86 @@ folly::Future<Status> ShowHostsExecutor::execute() {
...
+            r.emplace_back(leaderCount);
+            r.emplace_back(leaders.str());
+            r.emplace_back(parts.str());
+            v.emplace_back(std::move(r));
+        }   // row loop
+        return v;
+    };
+
+    auto makeGitInfoResult = [&](const std::vector<meta::cpp2::HostItem> &hostVec) -> DataSet {
+        DataSet v({"Host", "Port", "Status", "role", "git info sha"});
```
Comment (laura-ding) [inline src/executor/admin/ShowHostsExecutor.cpp:73]: "Host", "Port", "Status", "Role", "Git Info Sha"

Comment (liuyu85cn) [reply src/executor/admin/ShowHostsExecutor.cpp]: OK, fixed.

**Review by laura-ding — APPROVED** [review]

File: src/service/QueryEngine.cpp [hunk 3]
```diff
@@ -28,9 +29,12 @@ Status QueryEngine::init(std::shared_ptr<folly::IOThreadPoolExecutor> ioExecutor
     meta::MetaClientOptions options;
     options.serviceName_ = "graph";
     options.skipConfig_ = FLAGS_local_config;
-    metaClient_ = std::make_unique<meta::MetaClient>(ioExecutor,
-                                                     std::move(addrs.value()),
-                                                     options);
+    options.role_ = meta::cpp2::HostRole::GRAPH;
```
Comment (yixinglu) [inline src/service/QueryEngine.cpp:32]: Haven't role been deleted?

**Review by yixinglu — APPROVED** [review]

File: src/planner/Admin.h [hunk 4]
```diff
@@ -57,16 +57,21 @@ class DropNode : public SingleDependencyNode {
     bool ifExist_{false};
 };
 
-class ShowHosts final : public SingleDependencyNode {
+class ShowHosts final : public SingleInputNode {
```
Comment (CPWstatic) [inline src/planner/Admin.h:60]: Why change it to `SingleInput`?

Comment (CPWstatic) [reply src/planner/Admin.h:60]: `SingleDependencyNode ` is the right one.

Comment (liuyu85cn) [reply src/planner/Admin.h]: good catch, fixed.

File: src/service/QueryEngine.cpp (hunk 3 above)
Comment (liuyu85cn) [reply src/service/QueryEngine.cpp:32]: nope, not the same role

**Review by laura-ding — APPROVED** [review]

**Review by critical27 — APPROVED** [review]

**Pull request merged**